File: fc-chat/demo/part3/ch3/02-3-message-system-add-connection/src/main/java/net/prostars/messagesystem/dto/domain/InviteCode.java
package net.prostars.messagesystem.dto.domain;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;

public record InviteCode(@JsonValue String code) {

  @JsonCreator
  public InviteCode {
    if (code == null || code.isEmpty()) {
      throw new IllegalArgumentException("Invalid InviteCode");
    }
  }
}

File: fc-chat/demo/part3/with-push/src/main/kotlin/com/yscorp/withpush/messagesystem/dto/domain/InviteCode.kt
package com.yscorp.withpush.messagesystem.dto.domain

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonValue

@JvmRecord
data class InviteCode @JsonCreator constructor(@field:JsonValue val code: String?) {
    init {
        require(!(code == null || code.isEmpty())) { "Invalid InviteCode" }
    }
}
